entity-registry/src/main/java/com/linkedin/metadata/aspect/validation/ConditionalWriteValidator.java

```diff
@@ -0,0 +1,225 @@
+package com.linkedin.metadata.aspect.validation;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.net.HttpHeaders;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.events.metadata.ChangeType;
+import com.linkedin.metadata.aspect.AspectRetriever;
+import com.linkedin.metadata.aspect.RetrieverContext;
+import com.linkedin.metadata.aspect.SystemAspect;
+import com.linkedin.metadata.aspect.batch.BatchItem;
+import com.linkedin.metadata.aspect.batch.ChangeMCP;
+import com.linkedin.metadata.aspect.plugins.config.AspectPluginConfig;
+import com.linkedin.metadata.aspect.plugins.validation.AspectPayloadValidator;
+import com.linkedin.metadata.aspect.plugins.validation.AspectValidationException;
+import com.linkedin.metadata.aspect.plugins.validation.ValidationExceptionCollection;
+import com.linkedin.util.Pair;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.annotation.Nonnull;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+
+/**
+ * Implements conditional write semantics.
+ *
+ * <p>1. If an aspect change request includes a SystemMetadata section which includes a value for
+ * nextAspectVersion, then it must match the current SystemMetadata's value for nextAspectVersion or
+ * prevent performing the write operation.
+ *
+ * <p>2. Headers are included with the MCP for the following time-based checks * If-Modified-Since *
+ * If-Unmodified-Since
+ */
+@Setter
+@Getter
+@Accessors(chain = true)
+public class ConditionalWriteValidator extends AspectPayloadValidator {
```

Comment:
Is this validator first in the chain for validations?

Reply:
Not sure the validation order matters in this case, however the order is defined by the yaml file. All validations are processed before continuing. The sum of all validation exceptions are collected in the exception collection and returned to the user. So all validation errors are reported, there is no early exit function at the moment.

Reply:
Shouldn't this be first in the order and fail fast?